### PR TITLE
Add information from Lease to kubectl describe node

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/589

With this PR, the example output of `kubectl describe node bar` looks like:
```
	Name:               bar
        ...
        Unschedulable:      true
        Lease:
          HolderIdentity:  holder
          AcquireTime:     Mon, 14 Oct 2019 11:52:48 +0200
          RenewTime:       Mon, 14 Oct 2019 12:52:48 +0200
        Addresses:
        ...
```

```release-note
Add information from Lease object corresponding to a given Node to kubectl describe node output
```

@kubernetes/sig-node-pr-reviews

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go [hunk 1]
```diff
@@ -3000,6 +3001,15 @@ func (d *NodeDescriber) Describe(namespace, name string, describerSettings descr
 		return "", err
 	}
 
+	lease, err := d.CoordinationV1().Leases(corev1.NamespaceNodeLease).Get(name, metav1.GetOptions{})
```
Comment (wangzhen127) [inline staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go]: I remember there was a feature gate. Have we removed that? If not, maybe still guard this behind the feature gate?

File: staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go [hunk 2]
```diff
@@ -3043,6 +3053,24 @@ func describeNode(node *corev1.Node, nodeNonTerminatedPodsList *corev1.PodList,
 		w.Write(LEVEL_0, "CreationTimestamp:\t%s\n", node.CreationTimestamp.Time.Format(time.RFC1123Z))
 		printNodeTaintsMultiline(w, "Taints", node.Spec.Taints)
 		w.Write(LEVEL_0, "Unschedulable:\t%v\n", node.Spec.Unschedulable)
+
+		w.Write(LEVEL_0, "Lease:\n")
```
Comment (wangzhen127) [inline staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go]: Same here. Maybe put those prints behind feature gate?

File: staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go]: I don't think it's needed - we're going to GA it in 1.17 anyway and with GA the process is to just leave the feature gate itself but remove all checks on it.

So I'm going to leave as is. Note that if the feature is disabled, the only consequence is that we will write, e.g. "holder identity: <unset>", so I don't think it's actuallu a big deal.

File: staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go (hunk 2 above)
Comment (wojtek-t) [reply staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go]: Same answer as above.

**Comment by uthark** [comment]
Now regular user that has access to get/list nodes can't describe them.

```
kubectl describe node ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal
Error from server (Forbidden): leases.coordination.k8s.io "ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal" is forbidden: User "readonly.client.k8s" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "kube-node-lease"
```

Is this intended?

**Pull request merged**